File: packages/ui-swiftui/Sources/KinetixUI/Pagination.swift
//
// Pagination.swift — KinetixPagination family.
//
// Mirrors packages/ui/src/components/pagination.tsx, itself just
// `buttonVariants` recipes (Outline / Ghost, icon size). `size-9` (36) is
// off the shared spacing scale. Chevrons are `chevron.left` /
// `chevron.right` SF Symbols; the ellipsis is `ellipsis`.
//
// The React `PaginationContent` / `PaginationItem` wrappers (unstyled
// `<ul>` / `<li>`) aren't ported — the `KinetixPagination` HStack is the
// only container.
//

import SwiftUI

public struct KinetixPagination<Content: View>: View {
    private let content: Content

    public init(@ViewBuilder content: () -> Content) {
        self.content = content()
    }

    public var body: some View {
        HStack(spacing: 4) { content } // gap-1
            .accessibilityElement(children: .contain)
    }
}

public struct KinetixPaginationItem: View {
    @Environment(\.kinetixColors) private var colors

    private let label: String
    private let isActive: Bool
    private let action: () -> Void

    public init(_ label: String, isActive: Bool = false, action: @escaping () -> Void) {
        self.label = label
        self.isActive = isActive
        self.action = action
    }

    public var body: some View {
        Button(action: action) {
            Text(label)
                .font(.kinetixLabelLg)
                .foregroundStyle(isActive ? colors.foreground : colors.mutedForeground)
                .frame(width: 36, height: 36) // size-9
                .background(
                    isActive ? colors.accent : .clear,
                    in: RoundedRectangle(cornerRadius: 8, style: .continuous)
                )
                .overlay {
                    if isActive {
                        RoundedRectangle(cornerRadius: 8, style: .continuous)
                            .strokeBorder(colors.input, lineWidth: 1)
                    }
                }
                .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
        .accessibilityAddTraits(isActive ? [.isSelected] : [])
    }
}

public struct KinetixPaginationPrevious: View {
    @Environment(\.kinetixColors) private var colors
    private let action: () -> Void

    public init(action: @escaping () -> Void) {
        self.action = action
    }

    public var body: some View {
        Button(action: action) {
            HStack(spacing: 4) {
                Image(systemName: "chevron.left")
                Text("Previous")
            }
            .font(.kinetixLabelLg)
            .foregroundStyle(colors.foreground)
            .padding(.horizontal, 10)
            .frame(height: 36)
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
        .accessibilityLabel("Go to previous page")
    }
}

public struct KinetixPaginationNext: View {
    @Environment(\.kinetixColors) private var colors
    private let action: () -> Void

    public init(action: @escaping () -> Void) {
        self.action = action
    }

    public var body: some View {
        Button(action: action) {
            HStack(spacing: 4) {
                Text("Next")
                Image(systemName: "chevron.right")
            }
            .font(.kinetixLabelLg)
            .foregroundStyle(colors.foreground)
            .padding(.horizontal, 10)
            .frame(height: 36)
            .contentShape(Rectangle())
        }
        .buttonStyle(.plain)
        .accessibilityLabel("Go to next page")
    }
}

public struct KinetixPaginationEllipsis: View {
    @Environment(\.kinetixColors) private var colors

    public init() {}

    public var body: some View {
        Image(systemName: "ellipsis")
            .font(.kinetixBody)
            .foregroundStyle(colors.mutedForeground)
            .frame(width: 36, height: 36)
            .accessibilityLabel("More pages")
    }
}

File: packages/ui-flutter/lib/src/pagination.dart
import 'package:flutter/material.dart';

import 'app_text.dart';
import 'theme.dart';
import 'util.dart';

/// Mirrors `packages/ui/src/components/pagination.tsx`, itself
/// `buttonVariants` recipes. `size-9` (36) is off the shared spacing
/// scale. Chevrons are `chevron_left` / `chevron_right`; the ellipsis is
/// `more_horiz`.
class KinetixPagination extends StatelessWidget {
  const KinetixPagination({super.key, required this.children});

  final List<Widget> children;

  @override
  Widget build(BuildContext context) {
    return Row(
      mainAxisSize: MainAxisSize.min,
      children: gapAll(children, 4, axis: Axis.horizontal),
    );
  }
}

class KinetixPaginationItem extends StatelessWidget {
  const KinetixPaginationItem(this.label, {super.key, this.isActive = false, required this.onTap});

  final String label;
  final bool isActive;
  final VoidCallback onTap;

  @override
  Widget build(BuildContext context) {
    final c = KinetixTheme.of(context);
    return GestureDetector(
      onTap: onTap,
      behavior: HitTestBehavior.opaque,
      child: Container(
        width: 36,
        height: 36,
        alignment: Alignment.center,
        decoration: BoxDecoration(
          color: isActive ? c.accent : const Color(0x00000000),
          borderRadius: BorderRadius.circular(8),
          border: isActive ? Border.all(color: c.input, width: 1) : null,
        ),
        child: Text(
          label,
          style: AppText.labelLg.copyWith(color: isActive ? c.foreground : c.mutedForeground),
        ),
      ),
    );
  }
}

class KinetixPaginationPrevious extends StatelessWidget {
  const KinetixPaginationPrevious({super.key, required this.onTap});

  final VoidCallback onTap;

  @override
  Widget build(BuildContext context) {
    final c = KinetixTheme.of(context);
    return GestureDetector(
      onTap: onTap,
      behavior: HitTestBehavior.opaque,
      child: Container(
        height: 36,
        padding: const EdgeInsets.symmetric(horizontal: 10),
        alignment: Alignment.center,
        child: Row(
          mainAxisSize: MainAxisSize.min,
          children: [
            Icon(Icons.chevron_left, size: 16, color: c.foreground),
            const SizedBox(width: 4),
            Text(
              'Previous',
              style: AppText.labelLg.copyWith(color: c.foreground),
            ),
          ],
        ),
      ),
    );
  }
}

class KinetixPaginationNext extends StatelessWidget {
  const KinetixPaginationNext({super.key, required this.onTap});

  final VoidCallback onTap;

  @override
  Widget build(BuildContext context) {
    final c = KinetixTheme.of(context);
    return GestureDetector(
      onTap: onTap,
      behavior: HitTestBehavior.opaque,
      child: Container(
        height: 36,
        padding: const EdgeInsets.symmetric(horizontal: 10),
        alignment: Alignment.center,
        child: Row(
          mainAxisSize: MainAxisSize.min,
          children: [
            Text(
              'Next',
              style: AppText.labelLg.copyWith(color: c.foreground),
            ),
            const SizedBox(width: 4),
            Icon(Icons.chevron_right, size: 16, color: c.foreground),
          ],
        ),
      ),
    );
  }
}

class KinetixPaginationEllipsis extends StatelessWidget {
  const KinetixPaginationEllipsis({super.key});

  @override
  Widget build(BuildContext context) {
    final c = KinetixTheme.of(context);
    return SizedBox(
      width: 36,
      height: 36,
      child: Icon(Icons.more_horiz, size: 16, color: c.mutedForeground),
    );
  }
}
